### ISSUE TITLE
Tests do not work in Canada

### ISSUE THREAD

**Issue by mundya** [project-rig/rig#157]
I think the BMP tests have a hard-coded timestamp test.  This works in the UK apparently, but in Canada the 1970 that is expected becomes 1969. Oops.

**Comment by mossblaser** [comment]
Ooops indeed... However I can't find a test which checks a timestamp which isn't just expressed as a unix timestamp (i.e. a plain integer which doesn't go anywhere near a timezone-aware library). Could you paste in a copy of the failed test report?

**Comment by mossblaser** [comment]
Incidentally, it turns out that if the tests fail in Canada, they fail in real life!

![Canada](http://imgs.xkcd.com/comics/canada.png)

(Thanks @tomjnixon for making up for my recently lacklustre performance when it comes to relating everything back to the relevant XKCD...)

**Comment by mundya** [comment]
For bonus points one of our tests (for getting the machine, I think) assumes that core (3, 0, 17) is present. Sadly it's not in the SpiNN5 machine they have in Waterloo, so our test fails on that too. I guess that proves that cores that fail in Canada fail in real life too.

**Comment by mossblaser** [comment]
Why on earth are we testing that particular core? 17 is risky at the best
of times, never mind when in Canada!

**Comment by mundya** [comment]
Hmm, I can't find that now. I'll run the tests when I'm next to a board again.

**Comment by mundya** [comment]
```
=================================== FAILURES ===================================
__________________ TestMachineControllerLive.test_get_machine __________________

self = <test_machine_controller.TestMachineControllerLive object at 0x7faf038b89e8>
live_machine = <rig.machine.Machine object at 0x7faf038c41a8>
spinnaker_width = 8, spinnaker_height = 8

    def test_get_machine(self, live_machine, spinnaker_width,
                         spinnaker_height):
        # Just check that the output of get_machine is sane, doesn't verify
        # that it is actually correct. This test will fail if the target
        # machine is very dead...
        m = live_machine
    
        # This test will fail if the system has dead chips on its periphery
        assert m.width == spinnaker_width
        assert m.height == spinnaker_height
    
        # Check that *most* chips aren't dead or have resource exceptions
        assert len(m.chip_resource_exceptions) < (m.width * m.height) / 2
        assert len(m.dead_chips) < (m.width * m.height) / 2
        assert len(m.dead_links) < (m.width * m.height * 6) / 2
    
        # Check that those chips which are reported as dead are within the
        # bounds of the system
        for (x, y) in m.chip_resource_exceptions:
            assert 0 <= x < m.width
            assert 0 <= y < m.height
        for (x, y) in m.dead_chips:
            assert 0 <= x < m.width
            assert 0 <= y < m.height
            assert (x, y) not in m.chip_resource_exceptions
        for x, y, link in m.dead_links:
            assert 0 <= x < m.width
            assert 0 <= y < m.height
>           assert (x, y) not in m.chip_resource_exceptions
E           assert (3, 0) not in {(1, 4): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (2, 5): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (3, 0): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (3, 3): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, ...}
E            +  where {(1, 4): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (2, 5): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (3, 0): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, (3, 3): {Cores: 17, SDRAM: 119275520, SRAM: 24320}, ...} = <rig.machine.Machine object at 0x7faf038c41a8>.chip_resource_exceptions
```

**Comment by mossblaser** [comment]
Huh... :s well that test doesn't make any sense. Why should having a dead link imply not having resource exceptions! Whoopsie! It has been a bad few days for my tests ;)...

**Comment by mossblaser** [comment]
Fixed in #160.
